src/deploy/RPM_build/noobaa.spec

```diff
@@ -31,11 +32,40 @@ NooBaa is a data service for cloud environments, providing S3 object-store inter
 mkdir noobaa-core-%{version}-%{revision}
 mkdir node-%{nodever}
 tar -xzf %{SOURCE0} -C noobaa-core-%{version}-%{revision}/
-tar -xJf %{SOURCE1} -C node-%{nodever}/
 
 %clean
 [ ${RPM_BUILD_ROOT} != "/" ] && rm -rf ${RPM_BUILD_ROOT}
 
+%build
+#!/bin/bash
+NOOBAA_DIR="noobaa-core-%{version}-%{revision}"
+NODEJS_VERSION="%{nodever}"
+SKIP_NODE_INSTALL=1 source $NOOBAA_DIR/noobaa/src/deploy/NVA_build/install_nodejs.sh $NODEJS_VERSION
+
+pushd %{_sourcedir}
+nodepath=$(download_node)
+tar -xJf ${nodepath} -C %{_builddir}/node-%{nodever}/
+popd
+
+PATH=$PATH:%{_builddir}/node-%{nodever}/node-v$NODEJS_VERSION-linux-$(get_arch)/bin
+
+pushd $NOOBAA_DIR/noobaa
```

Comment:
No need to `pushd` here. You are already in the directory if you use `%setup` above.